src/onLCP.ts

```diff
@@ -58,6 +58,12 @@ export const onLCP = (onReport: LCPReportCallback, opts?: ReportOpts) => {
     let report: ReturnType<typeof bindReporter>;
 
     const handleEntries = (entries: LCPMetric['entries']) => {
+      // If reportAllChanges is set then call this function for each entry
+      // As otherwise only want to emit the last one.
+      if (opts!.reportAllChanges && entries.length > 1) {
+        entries.forEach((entry) => handleEntries([entry]));
```

Comment:
I think using a recursive call here make this code a bit hard to read.

Another way to do this would be to modify the `entries` array to ignore all but the last entry if `reportAllChanges` is false, and then do a `forEach()` over the logic below:

```
if (!opts.reportAllChanges) {
  entries = entries.slice(-1);
}
```

Reply:
Yeah I was trying to make the minimum number of calls so this, relatively rare, case had minimal impact on the more common not reportAllChanges case, but above is cleaner. Will do that.

Reply:
Done.